synapse/types.py

```diff
@@ -475,3 +478,24 @@ class ReadReceipt(object):
     user_id = attr.ib()
     event_ids = attr.ib()
     data = attr.ib()
+
+
+def get_verify_key_from_cross_signing_key(key_info):
+    """Get the key ID and signedjson verify key from a cross-signing key dict
+
+    Args:
+        key_info (dict): a cross-signing key dict, which must have a "keys"
+            property that has exactly one item in it
+
+    Returns:
+        (str, VerifyKey): the key ID and verify key for the cross-signing key
+    """
+    # make sure that exactly one key is provided
+    if "keys" not in key_info:
+        raise SynapseError(400, "Invalid key")
```

Comment:
Isn't this more of a ValueError than a SynapseError? I don't think it's up to a utility function to determine whether this translates into a 400 error at the API or not: that's a question for the REST layer or the handlers.

For example: the key passed into this function might have come out of the database. Where that is the case, the problem isn't with the request, but with the database, so the appropriate response is a 500, not a 400.